internal/pkg/queue/index/manager.go

```diff
@@ -57,22 +72,29 @@ func NewIndexManager(walPath, indexPath, queueDirPath string) (*IndexManager, er
 	}
 
 	im := &IndexManager{
-		hostIndex:    newIndex(),
-		walFile:      walFile,
-		indexFile:    indexFile,
-		queueDirPath: queueDirPath,
-		walEncoder:   gob.NewEncoder(walFile),
-		walDecoder:   gob.NewDecoder(walFile),
-		indexEncoder: gob.NewEncoder(indexFile),
-		indexDecoder: gob.NewDecoder(indexFile),
-		dumpTicker:   time.NewTicker(time.Duration(dumpFrequency) * time.Second),
-		lastDumpTime: time.Now(),
-		stopChan:     make(chan struct{}),
+		hostIndex:          newIndex(),
+		walFile:            walFile,
+		indexFile:          indexFile,
+		queueDirPath:       queueDirPath,
+		walEncoder:         gob.NewEncoder(walFile),
+		walDecoder:         gob.NewDecoder(walFile),
+		indexEncoder:       gob.NewEncoder(indexFile),
+		indexDecoder:       gob.NewDecoder(indexFile),
+		dumpTicker:         time.NewTicker(time.Duration(dumpFrequency) * time.Second),
+		lastDumpTime:       time.Now(),
+		walCommit:          new(atomic.Uint64),
+		walCommited:        new(atomic.Uint64),
+		walNotifyListeners: new(atomic.Int64),
+		walCommitedNotify:  make(chan uint64),
+		WAL_IO_PERCENT:     10,
+		WAL_MIN_INTERVAL:   10 * time.Millisecond,
+		walStopChan:        make(chan struct{}),
+		stopChan:           make(chan struct{}),
 	}
 
 	// Check if WAL file is empty
 	im.Lock()
-	empty, err := im.unsafeIsWALEmpty()
+	empty, err := im.unsafeIsWALEmpty() // FIXME: check error
```

Comment:
spotted FIXME

Reply:
https://github.com/internetarchive/Zeno/commit/7c6b1066729882a1c2a344cb4692c9962f8efe8f

I'm not sure about the intention here.

Reply:
you're right the error is not checked after return